--- bin/nfsend.orig	2017-09-06 12:36:19.000000000 -0400
+++ bin/nfsend	2017-09-06 12:57:26.000000000 -0400
@@ -789,7 +789,18 @@
 	# Check if configured sub hierarchy matches
 	CheckSubHierarchy($hints);
 
-	daemonize("$NfConf::PIDDIR/$pidfile");
+	if ( defined $NfConf::nodaemon && $NfConf::nodaemon ) {
+		unlink "$NfConf::PIDDIR/$pidfile";
+		open PID, ">$NfConf::PIDDIR/$pidfile" || die "Can't open pid file: $!\n";
+		print PID "$$\n";
+		if ( !close PID ) {
+			my $err = $!;
+			syslog('err', "Error startup nfsend: $err");
+			die "Unable to write PID file: $err";
+		}
+	} else {
+		daemonize("$NfConf::PIDDIR/$pidfile");
+	}
 
     if (!defined($comm_pid = fork)) {
         print STDERR "cannot fork: $!";
